### DISCUSSION TITLE
Option to specify what the top bar search is for

### DISCUSSION THREAD

**Discussion by muscanto** [snipe/snipe-it#9192] [Ideas]
Would it be possible to put a setting to allow us to choose what the search field in the top of the header bar is for?

I understand having it default to Asset tag, but it would be really nice for us to have it be assigned as a general search like you have at the top of the "List All" assets section.

**Comment by bra1ncramp** [comment]
I like this idea - more often we need to search for a person, and it's several clicks, backing out for a different user, etc. Being able to change the scope of the search (like a drop down for Asset tag, person, asset model, etc) would be nice.

**Comment by snipe** [comment]
This has been on the feature list for a while.

**Reply by muscanto** [reply]
Where do I adjust the setting?

**Reply by bra1ncramp** [reply]
@snipe , are you saying that this is currently available?

**Reply by snipe** [reply]
No, I meant it's been on the list of features we want to build.

**Comment by snipe** [comment]
So, the real issue here is the same as its always been.... that being an asset tag only search, the search result is effectively Boolean. There is either an exact match, or there is not. If there is not an exact match to that asset tag, it bumps you back to the assets listing page and tells you that asset tag doesn’t exist.

If we add additional search options... let’s say a drop down arrow that lets you decide what you’re searching on, something like below: 
 
![C2AD2313-ADAE-4492-B2C5-5ABCEAF209BA](https://user-images.githubusercontent.com/197404/114798062-5b028f80-9d49-11eb-88e3-22f8d8c8bccc.png)

![2549CFF7-62FD-4074-ADCB-2F5BBEB96980](https://user-images.githubusercontent.com/197404/114798063-5b9b2600-9d49-11eb-959f-e2c4de8ff9ff.png)

... when people search on partial things, what does that result screen look like? I *guess* we could try to funnel that server side request into the table results, but man that feels gross.

**Reply by bra1ncramp** [reply]
Could it possibly just pipe the search term into the search field on the respective page?

**Reply by muscanto** [reply]
99% of the time I am searching a users name or their computer's host name not the asset tag.   I'm not going to ask Jennifer in accounting in another state over the phone to undock her laptop and flip it over so she can give me the asset tag number if you know what I mean?  The search field at the top of the "list all" assets section works exactly how I would like the search field on the main screen to work.  I just want to avoid having to hit the assets section and then the list all section to get to the search field that actually does what I need for finding asset tags, users and hostnames all in one search.

**Reply by snipe** [reply]
Again, that top search bar is meant for barcode scanners - that was the entire point of that search bar, which is documented. 

> The search field at the top of the "list all" assets section works exactly how I would like the search field on the main screen to work.

They work completely differently by design though. (Literally, on the back end. One has a table of info already loaded and filters through, the other has to do its job irrespective of where you are in the app.)

This has always been a UX/UI issue, and I’m certain we’ve covered this in previous tickets.

I’m not saying we can’t do it, just that it’s a lot harder than it seems from a UX perspective.

**Reply by snipe** [reply]
Plus, if you have to select that arrow from the drop down like I’m my mockup, you’re not really saving yourself a ton of clicks.

**Reply by muscanto** [reply]
I see what you are saying on the limited function of the search.  I realized I've notice that limitation too.  If I search Jen Johnson when her name is Jennifer Johnson, the search fails.  If I search Jen, she comes up.  I'm not a code/programmer type but from a user experience viewpoint, that is certainly a limitation.  Are there any plans to improve this to account for spaces and broken words?  

Initially, I was more or less thinking that you could have a setting in the system menu that changes the behavior of what the search field on the main page does.  I was not thinking about a drop down on the main page as you are right, that would introduce more clicks.   I'm sure plenty of users use barcodes, we have them but never use them.  It's faster for us to just grab the tag number or use the hostname/username.

I understand the complications with the table not being loaded on the main page and how it handles the data.  That's certainly a hurdle but if you can come up with a create way to overcome some of these challenges, it would greatly improve the user experience in my opinion.  After all, we know how important good search is.. there's a reason google is king :)

**Reply by snipe** [reply]
There is, and Amazon's search sucks, and yet they're still a wildly successful company. These are hard problems to solve.
